lib/fika/env.ex

```diff
@@ -10,25 +10,25 @@ defmodule Fika.Env do
       # This is a map of known function signatures and their return types.
       # As a hack for now, we store kernel function signatures right here.
       function_types: %{
-        "kernel.+(Int,Int)" => "Int",
-        "kernel.+(Int,Float)" => "Float",
-        "kernel.+(Float,Int)" => "Float",
-        "kernel.+(Float,Float)" => "Float",
-        "kernel.-(Int,Int)" => "Int",
-        "kernel.-(Int,Float)" => "Float",
-        "kernel.-(Float,Int)" => "Float",
-        "kernel.-(Float,Float)" => "Float",
-        "kernel.*(Int,Int)" => "Int",
-        "kernel.*(Int,Float)" => "Float",
-        "kernel.*(Float,Int)" => "Float",
-        "kernel.*(Float,Float)" => "Float",
-        "kernel./(Int,Int)" => "Float",
-        "kernel./(Int,Float)" => "Float",
-        "kernel./(Float,Int)" => "Float",
-        "kernel./(Float,Float)" => "Float",
-        "kernel.|(Bool,Bool)" => "Bool",
-        "kernel.&(Bool,Bool)" => "Bool",
-        "kernel.!(Bool)" => "Bool"
+        "kernel.+(Int,Int)" => ["Int"],
+        "kernel.+(Int,Float)" => ["Float"],
+        "kernel.+(Float,Int)" => ["Float"],
+        "kernel.+(Float,Float)" => ["Float"],
+        "kernel.-(Int,Int)" => ["Int"],
+        "kernel.-(Int,Float)" => ["Float"],
+        "kernel.-(Float,Int)" => ["Float"],
+        "kernel.-(Float,Float)" => ["Float"],
+        "kernel.*(Int,Int)" => ["Int"],
+        "kernel.*(Int,Float)" => ["Float"],
+        "kernel.*(Float,Int)" => ["Float"],
+        "kernel.*(Float,Float)" => ["Float"],
+        "kernel./(Int,Int)" => ["Float"],
+        "kernel./(Int,Float)" => ["Float"],
+        "kernel./(Float,Int)" => ["Float"],
+        "kernel./(Float,Float)" => ["Float"],
+        "kernel.|(Bool,Bool)" => ["Bool"],
+        "kernel.&(Bool,Bool)" => ["Bool"],
+        "kernel.!(Bool)" => ["Bool"]
```

Comment:
It would be nice if we can use lists only for union types. Let's discuss this in discord if needed.

Reply:
I agree. My plan of action is:

- Incorporate the new atom parser
- Remove the lists when the type is not an union